ST: Add upgrade test for 0.17 and enhance jenkins pipeline

Frawless opened the pull request:
### Type of change

- Enhancement / new feature

### Description

This PR adds Strimzi 0.17 to upgrade chain system tests. I also add an enhancement for Jenkins job to use the proper image in upgrade tests in case that images were built locally during the build.

### Checklist

- [x] Make sure all tests pass

Frawless commented:
@strimzi-ci run tests profile=upgrade testcase=StrimziUpgradeST#testUpgradeStrimziVersion

Frawless commented:
@strimzi-ci run tests profile=upgrade testcase=StrimziUpgradeST#testUpgradeStrimziVersion

Frawless commented:
@strimzi-ci run tests profile=upgrade testcase=StrimziUpgradeST#testUpgradeStrimziVersion

Frawless commented:
@strimzi-ci run tests profile=upgrade testcase=StrimziUpgradeST#testUpgradeStrimziVersion

Frawless commented:
@strimzi-ci run tests profile=upgrade testcase=StrimziUpgradeST#testUpgradeStrimziVersion

Frawless commented:
@strimzi-ci run tests profile=upgrade testcase=StrimziUpgradeST#testUpgradeStrimziVersion

Frawless commented:
@strimzi-ci run tests profile=upgrade testcase=StrimziUpgradeST#testUpgradeStrimziVersion

Frawless commented:
@strimzi-ci run tests profile=upgrade testcase=StrimziUpgradeST#testUpgradeStrimziVersion

Frawless commented:
@strimzi-ci run tests profile=upgrade testcase=StrimziUpgradeST#testUpgradeStrimziVersion

Frawless commented:
@strimzi-ci run tests profile=upgrade testcase=StrimziUpgradeST#testUpgradeStrimziVersion

ppatierno commented on test/src/main/java/io/strimzi/test/k8s/cmdClient/BaseCmdKubeClient.java:
```diff
@@ -181,6 +181,8 @@ public K replace(File... files) {
     public K applyContent(String yamlContent) {
         try (Context context = defaultContext()) {
+            // TODO remove this
+            LOGGER.info(yamlContent);
```
what about this TODO to remove?

Frawless replied:
I am still trying to debug reason why last test failing. I will remove it later. Also I moved this PR to `draft`, because it's not ready net. Sorry for confusion

Frawless commented:
@strimzi-ci run tests profile=upgrade testcase=StrimziUpgradeST

Frawless commented:
@strimzi-ci run tests profile=upgrade testcase=StrimziUpgradeST

Frawless commented:
@strimzi-ci run tests profile=upgrade testcase=StrimziUpgradeST#testUpgradeStrimziVersion

Frawless commented:
@strimzi-ci run tests profile=upgrade testcase=StrimziUpgradeST#testUpgradeStrimziVersion

Frawless commented:
@strimzi-ci run tests profile=upgrade testcase=StrimziUpgradeST#testUpgradeStrimziVersion

Frawless commented:
@strimzi-ci run tests profile=upgrade testcase=StrimziUpgradeST

Frawless commented:
@strimzi-ci run tests profile=upgrade testcase=StrimziUpgradeST#testUpgradeStrimziVersion

Frawless commented:
So finally upgrade is working for PR builds. I added there option for flaky upgrade tests and removed k8s support options from `StrimziUpgradeST.json` It's ready for review.

tombentley commented on Jenkinsfile-pr:
```diff
@@ -130,6 +130,18 @@ pipeline {
                         if (env.OC_VERSION == "4" && !env.ghprbCommentBody.contains('test_only')) {
                             env.DOCKER_REGISTRY = "image-registry.openshift-image-registry.svc:5000"
                         }
+                        // For upgrade tests running with locally builded images we need to change tag in StrimziUpgradeST.json
```
```suggestion
                        // For upgrade tests running with locally buildt images we need to change tag in StrimziUpgradeST.json
```
